### PR TITLE
MGMT-4188 Use manifest API to manage operator ignition files

### PR DESCRIPTION
**Pull request by jordigilh**
Refactors the OLM operators manifest generation to use the `manifestAPI` to manage the files created instead of storing them locally into the file system.

@masayag @jakub-dzon @machacekondra @pkliczewski @gamli75 please review at your discretion.

Thanks,

/Jordi

### CODE REVIEW

File: internal/operators/cnv/manifest.go [hunk 1]
```diff
@@ -6,34 +6,34 @@ import (
 )
```
Comment (masayag) [inline internal/operators/cnv/manifest.go:6]: @jordigilh pls change the import package from `html/template` to `text/template`

File: internal/operators/manager.go [hunk 2]
```diff
@@ -69,17 +72,38 @@ func (mgr *Manager) GenerateManifests(cluster *common.Cluster) (map[string]strin
...
+		CreateManifestParams: &models.CreateManifestParams{
+			Content:  swag.String(base64.StdEncoding.EncodeToString(content)),
+			FileName: &filename,
+			Folder:   swag.String(models.ManifestFolderOpenshift),
+		},
+	})
+
+	if _, ok := response.(*operations.CreateClusterManifestCreated); !ok {
+		if apiErr, ok := response.(*common.ApiErrorResponse); ok {
+			return errors.Wrapf(apiErr, "Failed to create manifest %s", filename)
```
Comment (masayag) [inline internal/operators/manager.go:102]: pls include cluster id in the message to provide more context to the error.

File: internal/operators/cnv/manifest.go (hunk 1 above)
Comment (jordigilh) [reply internal/operators/cnv/manifest.go:6]: will do

File: internal/operators/manager.go [hunk 3]
```diff
@@ -69,17 +72,38 @@ func (mgr *Manager) GenerateManifests(cluster *common.Cluster) (map[string]strin
...
 			}
 		}
 	}
 
-	return operatorManifests, nil
+	return nil
+}
+
+func (mgr *Manager) createManifests(ctx context.Context, cluster *common.Cluster, filename string, content []byte) error {
+	// all relevant logs of creating manifest weill be inside CreateClusterManifest
```
Comment (masayag) [inline internal/operators/manager.go:90]: s/weill/will

File: internal/ignition/ignition.go [hunk 4]
```diff
@@ -244,12 +199,6 @@ func (g *installerGenerator) Generate(ctx context.Context, installConfig []byte)
 		}
 
 	}
-
-	err = g.writeOperatorManifests(installerPath, envVars)
```
Comment (masayag) [inline internal/ignition/ignition.go:248]: what will compensate for the removal of this call ? 
the manifests need to be added to the cluster before line 222 is reached

File: internal/operators/manager.go (hunk 2 above)
Comment (jordigilh) [reply internal/operators/manager.go:102]: :+1:

File: internal/operators/manager.go (hunk 3 above)
Comment (jordigilh) [reply internal/operators/manager.go:90]: indeed! :smile:

File: internal/ignition/ignition.go (hunk 4 above)
Comment (masayag) [reply internal/ignition/ignition.go:248]: Have a look at https://github.com/openshift/assisted-service/blob/master/internal/cluster/cluster.go#L830 - this is a place in which manifest files are being added internally during host installation

Comment (jordigilh) [reply internal/ignition/ignition.go:248]: I forgot to add the call to the operator's API. Thanks for pointing it out.

Comment (jordigilh) [reply internal/ignition/ignition.go:248]: > Have a look at https://github.com/openshift/assisted-service/blob/master/internal/cluster/cluster.go#L830 - this is a place in which manifest files are being added internally during host installation

yeah, that's where I was going to add it :smile:

**Pull request merged**